pkg/metrics/queue.go

```diff
@@ -177,6 +203,10 @@ func (m *QueueMetrics) IncQueueApplicationsRejected() {
 	m.incQueueApplications(AppRejected)
 }
 
+func (m *QueueMetrics) DecQueueApplicationsRejected() {
+	m.decQueueApplications(AppRejected)
+}
```

Comment:
DecQueueApplicationsRejected() is not called in "leave_Rejected" state? Is it intentional?

Reply:
Rejected applications should only increase, never decrease: a simple counter.
It is a final state that the application only gets removed from as part of the cleanup. We keep it around for a while to make it traceable for the submitter. i.e. submit an application that gets rejected: find it is the list as such. If we would drop it immediately the submitter has no idea what happened.

Reply:
@chenyulin0719 @wilfred-s 
If i make sense right, we don't need to add decrease for following state?
1. Rejected
2. Failed
3. Completed

And for expired, i don't think we need to tracking it?

Reply:
Addressed in latest PR.

Reply:
@wilfred-s Understood, thanks for the explanation.
@zhuqi-lucas Not tracking expired metrics make sense to me. 
To avoid any confusion, I think we should remove all the decrement function for Rejected/Failed/Completed.  WDYT?

Reply:
Got it @chenyulin0719 , remove those unused code in latest PR!